### PR TITLE
Refactor: turn `std::vector<LinkPair>` to separate class `Links`

### PR DESCRIPTION
**Pull request by bogdasar1985**
Issue https://github.com/newsboat/newsboat/issues/2464

According to this commit https://github.com/newsboat/newsboat/commit/0684f78c9b3f01258d23153ce70b37b4eb76763c I can't use `make depslist` anymore, so I add file `src/links.cpp` into `mk/newsboat.deps` manually in alphabetical order.

### CODE REVIEW

File: include/links.h [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#ifndef LINKS_H_
+#define LINKS_H_
+#include <vector>
+#include <utility>
+#include <string>
+namespace newsboat {
+// This enum has to be kept in sync with enum LinkType in rust/libnewsboat/src/htmlrenderer.rs
```
Comment (Minoru) [inline include/links.h:7]: Please also move rust/libnewsboat/src/htmlrenderer.rs to links.rs, update a comment in it to point to links.h, and update rust/libnewsboat/src/mod.rs to replace `htmlrenderer` with `links`.

File: include/links.h [hunk 2]
```diff
@@ -0,0 +1,61 @@
+#ifndef LINKS_H_
+#define LINKS_H_
```
Comment (Minoru) [inline include/links.h:2]: In Newsboat, we prefix all our include guards with `NEWSBOAT_`; please do that too.

File: src/htmlrenderer.cpp [hunk 3]
```diff
@@ -63,14 +63,14 @@ HtmlRenderer::HtmlRenderer(bool raw)
...
-	std::vector<LinkPair>& links,
+	Links& links,
 	const std::string& url)
 {
 	std::istringstream input(source);
 	render(input, lines, links, url);
 }
 
-unsigned int HtmlRenderer::add_link(std::vector<LinkPair>& links,
+unsigned int HtmlRenderer::add_link(Links& links,
```
Comment (Minoru) [inline src/htmlrenderer.cpp:73]: Can we delete this method, and replace calls to it with `links.add_link(..)`?

**Review by Minoru — CHANGES_REQUESTED** [review]
Looks pretty good!

Can you please add tests for the two invariants mentioned in https://github.com/newsboat/newsboat/issues/2464?

**Comment by bogdasar1985** [comment]
@Minoru I also wanna change `std::pair` to structure as you mentioned, because things like this `links.begin()->first` looks terrible. And `std::pair` is more about locally in-place usage (for example, iterate over `std::map`) and it's not our case.

**Review by Minoru — APPROVED** [review]
Excellent, thank you very much!

**Pull request merged**